frontend/pkg/frontend/middleware_tracing_test.go

```diff
@@ -0,0 +1,61 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/otel/baggage"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+)
+
+func TestContextWithTraceCorrelationData(t *testing.T) {
+	requestID := uuid.New()
+
+	expected := map[string]string{
+		"correlation.id":    "12345",
+		"client.request.id": "67890",
+		"request.id":        requestID.String(),
+	}
+
+	data := &arm.CorrelationData{
+		CorrelationRequestID: expected["correlation.id"],
+		ClientRequestID:      expected["client.request.id"],
+		RequestID:            requestID,
+	}
+
+	// NOTE: no span, no effect
```

Comment:
Just a nitpick. I think this unit test is fine but technically you test a few scenarios here. So I wonder if it would be more readable if you construct this into test cases:

``` suggestion
tests := []struct {
	name    string
	data	something
	wantErr string
}{
	{
		// test spec
	},
}

for _, tt := range tests {
	// do your test
}
```
Feel free to ignore this suggestion if it ends up overcomplicating the test.

Reply:
mh.. from point of view that would make it more compliated then its supposed to be.

Since 3 out of 4 tests, do just a simple function call with emtpy data or no active span to ensure, no active tracer will simply have no effect.